lib/rails_param/param_evaluator.rb

```diff
@@ -10,9 +10,7 @@ def param!(name, type, options = {}, &block)
       name = name.is_a?(Integer)? name : name.to_s
       return unless params.include?(name) || check_param_presence?(options[:default]) || options[:required]
 
-      # coerce value (but only if a value was provided)
-      coerced_value = params.include?(name) ?
```

Comment:
I have moved testing for this into the coerce method `params[name]` will return `nil` if the value was provided as `nil`, and if the value was not provided at all.